skbio/alignment/_alignment.py

```diff
@@ -72,13 +72,16 @@ def __init__(self, seqs):
         self._data = seqs
         self._id_to_index = {}
         for i, seq in enumerate(self._data):
-            id = seq.id
-            if id in self:
+            if 'id' not in seq.metadata:
```

Comment:
That's not desirable behavior
On Jun 9, 2015 5:27 PM, "Anders Pitman" <notifications@github.com> wrote:

> In skbio/alignment/_alignment.py
> <https://github.com/biocore/scikit-bio/pull/938#discussion_r32079140>:
>
> > @@ -72,13 +72,16 @@ def __init__(self, seqs):
> >          self._data = seqs
> >          self._id_to_index = {}
> >          for i, seq in enumerate(self._data):
> > -            id = seq.id
> > -            if id in self:
> > +            if 'id' not in seq.metadata:
>
> @jairideout <https://github.com/jairideout> @ebolyen
> <https://github.com/ebolyen> it appears there is precedent for adding the
> ids if they don't exist. See for example this test
> <https://github.com/anderspitman/scikit-bio/blob/1b41d2a6339ac4f0772db31b1345bc3587f0c7e8/skbio/alignment/tests/test_pairwise.py#L289>
> .
>
> Thoughts?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/biocore/scikit-bio/pull/938/files#r32079140>.
>